pyccel/ast/builtins.py

```diff
@@ -417,7 +417,15 @@ def __init__(self, *args):
             self._shape     = (LiteralInteger(len(args)), ) + args[0].shape
 
     def __getitem__(self,i):
-        return self._args[i]
+        if isinstance(i, (int, LiteralInteger)):
+            return self._args[i]
```

Comment:
Shouldn't we also consider the case of a negative literal index here? Something like `PyccelUnarySub(LiteralInteger(1))`.

Maybe we need to provide the `__index__` method to `PyccelUnarySub`, and then simply check with
```suggestion
        if hasattr(i, '__index__'):
            return self._args[i]
```

Reply:
You are right that we also need to consider a negative literal index, but I don't think we can provide the `__index__` method to `PyccelUnarySub` as there is no guarantee that the argument of `PyccelUnarySub` is a literal

Reply:
I think that `__index__` should return an `int` if it can, otherwise just `self` which will cause a `TypeError`.

Reply:
Sorry, my suggestion makes no sense because `PyccelUnarySub` should be passed to `IndexedVariable` in this case...

Reply:
Maybe we need an `eval` method? And then we can check if it returns an `int`?

Reply:
Perhaps, but in this case I think this ought to be in all operators and is a pull request of its own. For now I am handling this locally: bdecf75

Reply:
Sounds good